### PR TITLE
crush/CrushLocation: remove useless flag

### PR DESCRIPTION
**Pull request by rosinL**
In pr ‘https://github.com/ceph/ceph/pull/51352’, we have
crimson/crush/CrushLocation.cc, there is no need crimson
releated code in crush/CrushLocation.cc, so just remove.

Signed-off-by: luo rixin <luorixin@huawei.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by rosinL** [comment]
jenkins retest this please

File: src/crush/CrushLocation.cc [hunk 1]
```diff
@@ -93,9 +91,8 @@ int CrushLocation::update_from_hook()
 
   std::string out;
   bl.begin().copy(bl.length(), out);
-  out.erase(out.find_last_not_of(" \n\r\t")+1);
+  boost::algorithm::trim_if(out, boost::algorithm::is_any_of(" \n\r\t"));
```
Comment (athanatos) [inline src/crush/CrushLocation.cc:94]: I don't think these have the same behavior.  What is the behavior you expect?

**Review by athanatos — CHANGES_REQUESTED** [review]
See above

File: src/crush/CrushLocation.cc (hunk 1 above)
Comment (rosinL) [reply src/crush/CrushLocation.cc:94]: yeah, a little different, `trim_if` will trim both the left and right side of the string.
e.g.
```
string out= "\n\r0\n\r123\n\r"
out.erase(out.find_last_not_of(" \n\r\t")+1); //out is "\n\r0\n\r123"
boost::algorithm::trim_if(out, boost::algorithm::is_any_of(" \n\r\t")); // out is "0\n\r123"
```
Does it need to change `boost::algorithm::trim_if` to `boost::algorithm::trim_right_if`?

Comment (athanatos) [reply src/crush/CrushLocation.cc:94]: Oh, I actually misunderstood find_last_not_of.  Yeah, boost::algorithm::trim_right_if should be perfect.

**Review by athanatos — APPROVED** [review]

**Comment by rosinL** [comment]
jenkins retest this please

**Comment by rosinL** [comment]
jenkins retest this please

**Comment by rosinL** [comment]
jenkins test make check arm64

**Comment by ljflores** [comment]
Hey @rosinL, this is now getting tested!

**Comment by rosinL** [comment]
thank you @ljflores

**Pull request merged**